pkg/controller/nodeconfig/sync.go

```diff
@@ -103,34 +111,160 @@ func (ncc *Controller) sync(ctx context.Context, key string) error {
 	}
 
 	var errs []error
+	err = controllerhelpers.RunSync(
+		&status.Conditions,
+		namespaceControllerProgressingCondition,
+		namespaceControllerDegradedCondition,
+		nc.Generation,
+		func() ([]metav1.Condition, error) {
+			return ncc.syncNamespaces(ctx, nc, namespaces)
+		},
+	)
+	if err != nil {
+		errs = append(errs, fmt.Errorf("can't sync Namespace(s): %w", err))
+	}
 
-	err = ncc.syncNamespaces(ctx, namespaces)
+	err = controllerhelpers.RunSync(
+		&status.Conditions,
+		clusterRoleControllerProgressingCondition,
+		clusterRoleControllerDegradedCondition,
+		nc.Generation,
+		func() ([]metav1.Condition, error) {
+			return ncc.syncClusterRoles(ctx, nc, clusterRoles)
+		},
+	)
 	if err != nil {
-		errs = append(errs, fmt.Errorf("sync Namespace(s): %w", err))
+		errs = append(errs, fmt.Errorf("can't sync ClusterRole(s): %w", err))
 	}
 
-	err = ncc.syncClusterRoles(ctx, clusterRoles)
+	err = controllerhelpers.RunSync(
+		&status.Conditions,
+		serviceAccountControllerProgressingCondition,
+		serviceAccountControllerDegradedCondition,
+		nc.Generation,
+		func() ([]metav1.Condition, error) {
+			return ncc.syncServiceAccounts(ctx, nc, serviceAccounts)
+		},
+	)
 	if err != nil {
-		errs = append(errs, fmt.Errorf("sync ClusterRole(s): %w", err))
+		errs = append(errs, fmt.Errorf("can't sync ServiceAccount(s): %w", err))
 	}
 
-	err = ncc.syncServiceAccounts(ctx, serviceAccounts)
+	err = controllerhelpers.RunSync(
+		&status.Conditions,
+		clusterRoleBindingControllerProgressingCondition,
+		clusterRoleBindingControllerDegradedCondition,
+		nc.Generation,
+		func() ([]metav1.Condition, error) {
+			return ncc.syncClusterRoleBindings(ctx, nc, clusterRoleBindings)
+		},
+	)
 	if err != nil {
-		errs = append(errs, fmt.Errorf("sync ServiceAccount(s): %w", err))
+		errs = append(errs, fmt.Errorf("can't sync ClusterRoleBinding(s): %w", err))
 	}
 
-	err = ncc.syncClusterRoleBindings(ctx, clusterRoleBindings)
+	err = controllerhelpers.RunSync(
+		&status.Conditions,
+		daemonSetControllerProgressingCondition,
+		daemonSetControllerDegradedCondition,
+		nc.Generation,
+		func() ([]metav1.Condition, error) {
+			return ncc.syncDaemonSet(ctx, nc, soc, daemonSets)
+		},
+	)
 	if err != nil {
-		errs = append(errs, fmt.Errorf("sync ClusterRoleBinding(s): %w", err))
+		errs = append(errs, fmt.Errorf("can't sync DaemonSet(s): %w", err))
 	}
 
-	err = ncc.syncDaemonSet(ctx, nc, soc, daemonSets)
+	// Aggregate conditions.
+	var aggregationErrs []error
+
+	availableConditions := controllerhelpers.FindStatusConditionsWithSuffix(status.Conditions, scyllav1alpha1.AvailableCondition)
+
+	// Use a default available node condition for nodes which haven't propagated their conditions to status yet.
```

Comment:
I extracted a func creating default node conditions. I could probably go further but imo it would make it more convoluted than it's worth it. Let me know what you think.